### PR TITLE
Fix zone timer on reconnect

### LINKED ISSUE #818: [BUG] Zone time on reconnect
## 🐛 Bug Report

When you reconnect to a game , the zone timer starts again. This is because we are mapping the GameSettings.json `map_shrink_wait_ms` value. We need to track the current zone time of the game.

### PR DESCRIPTION
**Pull request by tkz00**
Registered the start time of the match, calculate based on that the time elapsed since the beginning of it, subtract from the time it takes to start shrinking the zone the time elapsed and display it.

Closes #818

### CODE REVIEW

**Review by BertovDev — CHANGES_REQUESTED** [review]
When timer reaches 0:

Captura de pantalla 2023-10-23 a la(s) 14 44 43

File: client/Assets/Scripts/UI/MatchStatsController.cs
Comment (BertovDev) [inline client/Assets/Scripts/UI/MatchStatsController.cs:1]: LGTM! consider doing something like this to improve code readability:

```
        ulong mapShrinkWaitMs = LobbyConnection
            .Instance
            .serverSettings
            .RunnerConfig
            .MapShrinkWaitMs;
```

Comment (tkz00) [reply client/Assets/Scripts/UI/MatchStatsController.cs:1]: Looks good! Already made the changes!

File: client/Assets/Scripts/UI/MatchStatsController.cs [hunk 1]
```diff
@@ -39,13 +25,22 @@ void FixedUpdate()
...
+                .serverSettings
+                .RunnerConfig
+                .MapShrinkWaitMs;
+            if(mapShrinkWaitMs > elapsedTime) {
+                remainingSeconds = (mapShrinkWaitMs - elapsedTime) / 1000;
+                zoneTimer.text = remainingSeconds.ToString();
+            }
+            else {
+                remainingSeconds = 0;
+                zoneTimer.text = "0";
```
Comment (ncontinanza) [inline client/Assets/Scripts/UI/MatchStatsController.cs:42]: In both cases, having `zoneTimer.text = remainingSeconds.ToString();` will do. What do you think of leaving that line only once outside de if/else? After that, maybe you can go further and rewrite the assignment as a single statement using ternary operation if you feel it more readable/cleaner.

**Review by ncontinanza — CHANGES_REQUESTED** [review]
Working great! Just a little comment about the code

File: client/Assets/Scripts/SocketConnectionManager.cs [hunk 2]
```diff
@@ -163,6 +163,11 @@ private void OnWebSocketMessage(byte[] data)
                     this.gameProjectiles = gameEvent.Projectiles.ToList();
                     alivePlayers = gameEvent.Players.ToList().FindAll(el => el.Health > 0);
                     updatedLoots = gameEvent.Loots.ToList();
+
+                    if(LobbyConnection.Instance.matchStartTime == 0) {
+                        LobbyConnection.Instance.matchStartTime = (ulong)gameEvent.ServerTimestamp;
+                        PlayerPrefs.SetString("matchStartTime", ((ulong)gameEvent.ServerTimestamp).ToString());
```
Comment (saugon) [inline client/Assets/Scripts/SocketConnectionManager.cs:169]: This approach to solving the issue could be appropriate only if it weren't possible to make any changes on the backend side.

In our case, the best approach is to obtain the current time directly from the backend and display it without requiring any calculations on the client side.

Comment (tkz00) [reply client/Assets/Scripts/SocketConnectionManager.cs:169]: That's right, it should be the backend's responsibility to calculate and return that information, I'll close this PR and we are going to fix the issue with the corresponding backend and client changes in a new PR.

**Comment by tkz00** [comment]
After some consideration we came to the conclusion the best approach to display the remaining time for the damage zone would be to return the number from the backend, instead of calculating it in the front end, so we'll close this PR and fix the problem properly in a future one.

**Pull request closed without merging**